conda/core/solve.py

```diff
@@ -518,7 +518,8 @@ def _package_has_updates(self, ssc, spec, installed_pool):
                     has_update = True
                     break
         # let conda determine the latest version by just adding a name spec
-        return MatchSpec(spec.name) if has_update else None
+        return (MatchSpec(spec.name, version=prec.version, build_number=prec.build_number)
```

Comment:
This super-specific version/build number doesn't matter for solving. We used it as a reference point to say "can you install this?  If not, which specs are in the way?" and we then unfreeze them if we can.